airbyte-integrations/connectors/source-harvest/source_harvest/streams.py

```diff
@@ -0,0 +1,296 @@
+    pass
+
+
+class Estimates(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class EstimateMessages(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    parent_stream_name = "Estimates"
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"estimates/{stream_slice['parent_id']}/messages"
+
+
+class EstimateItemCategories(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Expenses(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class ExpenseCategories(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Tasks(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class TimeEntries(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class UserAssignments(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class TaskAssignments(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Projects(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Roles(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    pass
+
+
+class Users(HarvestStreamWithPagination, HarvestStreamIncrementalMixin):
+    def path(self, **kwargs) -> str:
+        return self.name
+
+
+class BillableRates(HarvestStreamWithPaginationSliced):
+    parent_stream_name = "Users"
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/billable_rates"
+
+
+class CostRates(HarvestStreamWithPaginationSliced):
+    parent_stream_name = "Users"
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/cost_rates"
+
+
+class ProjectAssignments(HarvestStreamWithPaginationSliced, HarvestStreamIncrementalMixin):
+    parent_stream_name = "Users"
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"users/{stream_slice['parent_id']}/project_assignments"
+
+
+class ReportsBase(HarvestStreamWithPagination, ABC):
+    data_field = "results"
+    date_param_template = "%Y%m%d"
+
+    def __init__(self, from_date: pendulum.date = pendulum.now().date().subtract(years=1), **kwargs):
+        super().__init__(**kwargs)
+        self._from_date = from_date
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(**kwargs)
+        current_date = pendulum.now()
+        # `from` and `to` params are required for expenses reports calls
+        # min `from` value is current_date - 1 year
+        params.update({"from": self._from_date.strftime("%Y%m%d"), "to": current_date.strftime("%Y%m%d")})
+        return params
+
+
+class ExpensesClients(ReportsBase):
+    def path(self, **kwargs) -> str:
```

Comment:
you can have a default implementation of `path` in the base class, and have something like `report_path="expenses/clients` here